### PR TITLE
feat: click to copy message

### PR DESCRIPTION
**Pull request by byBackfish**
Adds a component to every message, that on click copies that message content to the clipboard.
Pressing `L_CONTROL` whilst clicking on the component copies the raw message, including the color codes

### CODE REVIEW

File: src/main/java/com/wynntils/modules/chat/events/ClientEvents.java [hunk 1]
```diff
@@ -4,13 +4,15 @@
 
 package com.wynntils.modules.chat.events;
 
+import com.mojang.realmsclient.gui.ChatFormatting;
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/chat/events/ClientEvents.java]: Is this used?

Comment (byBackfish) [reply src/main/java/com/wynntils/modules/chat/events/ClientEvents.java]: yea, it is used for `ChatFormatting#stripFormatting`, I know that `TextFormatting#getTextWithoutFormattingCodes` also exists, but I have always used `ChatFormatting` for that. If you want, then i can change it to use `TextFormatting` instead

Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/chat/events/ClientEvents.java]: We use ChatFormatting in 1.18.2 because it is renamed there.

Comment (byBackfish) [reply src/main/java/com/wynntils/modules/chat/events/ClientEvents.java]: Yea, I've alway just used `ChatFormatting` in my previous mods, and never thought about switching to `TextFormatting`. I'll change it

File: src/main/java/com/wynntils/modules/chat/events/ClientEvents.java [hunk 2]
```diff
@@ -183,4 +191,28 @@ public void onClickNPC(PacketEvent<CPacketUseEntity> e) {
         ChatManager.progressDialogue();
     }
 
+    @SubscribeEvent
+    public void onMessageReceived(ClientChatReceivedEvent e) {
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/chat/events/ClientEvents.java:195]: I think this does make every message with a hover or click effect sent from Wynncraft lose their events. Did you test that?

Comment (byBackfish) [reply src/main/java/com/wynntils/modules/chat/events/ClientEvents.java:195]: Pretty sure it does not, testing once more just to make sure. It only adds a component, and does not change the existing ones

Comment (byBackfish) [reply src/main/java/com/wynntils/modules/chat/events/ClientEvents.java:195]: Yep, seems to work just fine

Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/chat/events/ClientEvents.java:195]: Ah, I miss read the code. But. Make this toggle able and off by default

**Review by HighCrit — APPROVED** [review]

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**